pkg/policy/mutate/validate.go

```diff
@@ -1,63 +1,54 @@
 package mutate
 
 import (
-	"errors"
 	"fmt"
 
 	kyverno "github.com/kyverno/kyverno/api/kyverno/v1"
-	commonAnchors "github.com/kyverno/kyverno/pkg/engine/anchor/common"
-	"github.com/kyverno/kyverno/pkg/policy/common"
 )
 
 // Mutate provides implementation to validate 'mutate' rule
 type Mutate struct {
-	// rule to hold 'mutate' rule specifications
-	rule kyverno.Mutation
+	mutation kyverno.Mutation
 }
 
 //NewMutateFactory returns a new instance of Mutate validation checker
-func NewMutateFactory(rule kyverno.Mutation) *Mutate {
-	m := Mutate{
-		rule: rule,
+func NewMutateFactory(m kyverno.Mutation) *Mutate {
+	return &Mutate{
+		mutation: m,
 	}
-	return &m
 }
 
 //Validate validates the 'mutate' rule
 func (m *Mutate) Validate() (string, error) {
-	rule := m.rule
-	// JSON Patches
-	if len(rule.Patches) != 0 {
-		for i, patch := range rule.Patches {
-			if err := validatePatch(patch); err != nil {
-				return fmt.Sprintf("patch[%d]", i), err
-			}
-		}
-	}
-	// Overlay
-	if rule.Overlay != nil {
-		path, err := common.ValidatePattern(rule.Overlay, "/", []commonAnchors.IsAnchor{commonAnchors.IsConditionAnchor, commonAnchors.IsAddingAnchor})
-		if err != nil {
-			return path, err
-		}
+	if m.hasForEach() {
```

Comment:
How do we validate other types of rules? i.e., we cannot have multiple mutate patterns in the same rule.

Reply:
Yes, this was missing before. Will add.